File: lab/lab01/clojure/src/circle_area.clj
;;;; circle_area.clj calculates the area of a circle.
;;;;
;;;; Input: The radius of a circle.
;;;; Output: The area of that circle.
;;;;
;;;; Usage: clojure -m circle_area
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by: Hansol Kim (hk94)
;;;; Date: January 13, 2022
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns circle_area) ; namespace function names the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function circleArea() computes the area of a circle.
;;; Receive: itsRadius, a number.
;;; Precondition: itsRadius >= 0.0.
;;; Return: the area of the corresponding circle.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn circleArea [itsRadius]
  (* Math/PI (* itsRadius itsRadius) )  ; return PI*r^2
)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main is where execution begins
;;; Input: the radius of a circle.
;;; Output: the area of that circle.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main [] ; defines where the execution begins. 
  (println "\nTo compute the area of a circle,") ; prints out strings with a new line at the bottom. 
  (print   " enter its radius: ") (flush) ; prints out the string. 
  (let
    [ radius (read) ] 

    (assert (>= radius 0) "-main: radius must be positive") ; proceed if the input radius is apositive number. 
    (printf "\nThe area is %f\n\n" (circleArea radius)) ; method circleArea calculates the radius in float and displays out one the console.  

    (print "\nThe area is ") 
    (print (circleArea radius)) 
    (print "\n\n")

    (printf "\nThe area is %.15f\n\n" (circleArea radius))
  )
) 

File: lab/lab01/circle_area.rb
#! /usr/bin/ruby
# circle_area.rb computes the area of a circle given its radius
# 
# Input: the radius of a circle
# Precondition: the radius is not negative
# Output: the area of the circle.
#
# Begun by: Prof. Adams, for CS 214 at Calvin College.
# Completed by: Hansol Kim (hk94)
# Date: January 13, 2022
###############################################################

# function circleArea returns a circle's area, given its radius
# Parameters: r, a number
# Precondition: r > 0.
# Returns: the area of a circle whose radius is r.
PI = 3.1415927 # Set a pi number
def circleArea(r) # Defines a function with a parameter r that stands for radius. 
    PI * r ** 2 # Make an equation using the set number pi and radius. Radius can change depeding on the user's input. 
end # Closing block of the method. 

if __FILE__ == $0 # Reference to the current filename. 
   puts "To compute the area of a circle," # puts can both display the calculated result and strings on the console. 
   print " enter its radius: " # print can only print out strings on the console. 
   radius = gets.chomp.to_f # gets input from the user and store in radius and to_f means to float. 
   print "The circle's area is: " # prints out the strings. 
   puts circleArea(radius) # prints out the calculated circle area from the circleArea function by receiving radius from the user. 
end # Closing block of the statement. 